src/OpenTelemetry/Trace/Export/BatchingActivityProcessor.cs

```diff
@@ -188,21 +182,74 @@ protected virtual void Dispose(bool isDisposing)
                     }
                     catch (Exception e)
                     {
-                        OpenTelemetrySdkEventSource.Log.SpanProcessorException("Dispose", e);
+                        OpenTelemetrySdkEventSource.Log.SpanProcessorException(nameof(this.Dispose), e);
                     }
                 }
+
+                this.flushTimer.Dispose();
+                this.flushLock.Dispose();
+                this.isDisposed = true;
             }
         }
 
-        private async Task ExportBatchAsync(CancellationToken cancellationToken)
+        private async Task FlushAsyncInternal(bool drain, CancellationToken cancellationToken)
         {
+            await this.flushLock.WaitAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Is it possible to unit test the scenario where multiple Flush is called from the timer itself? i.e one Timer event is still not finished (slooowww export) and the next one is fired?

Reply:
Check out my comment about the timer. Given that behavior, you think it's still needed?

Reply:
which comment?

Reply:
Hah forgot to send. Should be there now.

Reply:
makes sense.

Reply:
Does the timer mechanism now keeps firing at regular, fixed intervals? or the timer "stop counter" when Enabled = false;?

Reply:
if interval is d, then do we always export at T, T+d, T+2d? or T, T+(flush duration1)+d, T + (flush duration2) + d and so on.

Reply:
@cijothomas , that would be true unless the flush takes more then d time. Because only after flushing it would enable again.